### ISSUE TITLE
`SearchClient` ignores `sendApiKeyAsQueryParam=true` and send the api key in the query string

### ISSUE THREAD

**Issue by janaka** [typesense/typesense-js#101]
## Description

When the `SearchClient` is instantiated with a `ConfigurationsOptions` object container `sendApiKeyAsQueryParam=true` the query string param`x-typesense-api-key` still appears in the outgoing request. 

## Steps to reproduce

I have the following two snippets in a React app. 

```typescript
  const tsSearchClient = new TypesenseSearchClient({
    nodes: [
      {
        host: <typesene_host>, 
        port: <port>,
        protocol: 'http',
      },
    ],
    sendApiKeyAsQueryParam: false,
    apiKey: <typesense_api_key',
    connectionTimeoutSeconds: 2,
  });
```

```typescript
let response = await typesenseClient.multiSearch.perform(searchReq(), commonSearchParams(queryInput));
```

Pull up Chrome dev tools, network tab. You'll be able to see the param in the request URL. 

## Expected Behavior

The default appears to be `sendApiKeyAsQueryParam=false` in the implementation which makes sense to me. Unless explicitly set to `true` I expect the key to _NOT_ appear in the request URL.

## Actual Behavior

Regardless of the attribute being set to true, false, or not present the key appear in the request URL.

## Metadata

**Typsense Version**: 1.1.3-0

**OS**:
macOS Monterey 12.1
Chrome: Version 96.0.4664.110 (Official Build) (x86_64)

**Comment by janaka** [comment]
Thanks - will try getting to this today or tomorrow.

**Comment by janaka** [comment]
Hey @jasonbosco, sorry, I've not managed to teat this properly yet. I haven't had much time on this project since and been focused on refactoring a few bits to for the "production" setup.

**Comment by joeypunzel** [comment]
Thanks @janaka for submitting this issue - we just came across the same. 

@jasonbosco We tested the solution on branch `fix-sendApiKeyAsQueryParam` and can confirm this works as expected! We apply the sendApiKeyAsQueryParam=true on the TypesenseSearchClient object and requests go out with the API key omitted (see screenshot below). 

Any chance you guys can release this feature into the master branch soon? 

![image](https://user-images.githubusercontent.com/5006941/179313158-6950dd1e-fa2f-4a05-bed8-7375a966f798.png)
